src/typeChecker/__tests__/source4TypedModules.test.ts

```diff
@@ -0,0 +1,287 @@
+import { mockContext } from '../../mocks/context'
+import { Chapter, Variant } from '../../types'
+import { parse } from '../../parser/parser'
+import { parseError } from '../../index'
+
+let context = mockContext(Chapter.SOURCE_4, Variant.TYPED)
+
+beforeEach(() => {
+  context = mockContext(Chapter.SOURCE_4, Variant.TYPED)
+  context.nativeStorage.loadedModules = {
+    module: {}
```

Comment:
Nit: let's name it `exampleModule` or some other thing so that first time developers don't get confused